cli/src/main/java/org/jboss/gm/cli/Main.java

```diff
@@ -227,6 +218,12 @@ private void executeGradle(OutputStream out, boolean quiet, List<String> current
             logger.info("Executing CLI {} on Gradle project {} with JVM args '{}' and arguments '{}'",
                     ManifestUtils.getManifestInformation(Main.class), target, jvmArgs, currentGradleArgs);
 
+            if (logger.isDebugEnabled()) {
+                Map<String, String> envVarsToPrint = new HashMap<>(envVars);
+                envVarsToPrint.keySet().removeIf(k -> k.matches("^(LS_COLORS|RPROMPT|PROMPT|PS1)$"));
+                logger.debug("Environment variables: {}", envVarsToPrint);
+            }
```

Comment:
This isn't quite what I meant - I meant remove those aforementioned variables completely and avoid passing them through to Gradle - which then also reduces the debug output.

Reply:
Sorry, I thought you were referring to the log output only in your comment. Why do we want to filter any of the variables passed in, which depending on the environment may not even be set in the first place?

One thing I am trying to do with this PR is make launching Gradle from `Main` the same as if it was launched from the command-line. From the command line, it can apparently see all environment variables.

I originally thought it had something to do with forking/daemon, but that didn't seem to change the behavior.

Reply:
That's sort of the inverse of the problem in original fix which only passed in a few particular environment variables and dropped all others. This means that our process launched from `Main` cannot see what the build itself will see even if we run it on the same system one right after the other.

Reply:
According to the Gradle code, it does not print the values as they main contain sensitive information. This should also solve the too much output issue.

Reply:
I'll try this tomorrow but at first glance if its the same as Gradle internals then it should be ok. My only concern is older 4.10 runs and if our prompts contain non-ascii symbols - I'll try it tomorrow.

Reply:
I think I originally had tried this in the test. I will update the test. It's just make any variable contain non-ASCII symbols, right?

Reply:
I've tried it with an older build (fge/btf.git) and using 4.10.2 e.g.

```
java -jar ~/Work/Gradle/gradle-manipulator/cli/build/libs/cli-3.7-SNAPSHOT.jar -d  --init-script=~/Work/Gradle/gradle-manipulator/analyzer/build/resources/main/analyzer-init.gradle -DdependencySource=NONE generateAlignmentMetadata -l=~/Software/gradle-4.10.2
```
With a prompt containing symbols e.g. `%F{red}🎩%` running the above crashes out before even processing the build. Removing the prompt/etc variables it at least processes the build, but then the daemon crashes out at the end (which is what it did before).

Reply:
I had to rearrange some code to reuse the code that was already there for the version check. I updated the log message to also include the URL to the upstream issue, if that's OK.

Sorting is done exactly as suggested here.